models/product.go

```diff
@@ -25,26 +25,26 @@ type UserProducts struct {
 }
 type ProductUsers map[uuid.UUID]int
 
-func (l Privileges) IsValidPrivilege(privilege int) bool {
-	for _, value := range l {
+func (l *Privileges) IsValidPrivilege(privilege int) bool {
+	for _, value := range *l {
```

Comment:
Now I see why you not used pointer receiver here. If you use a pointer receiver, you should check for nilness before unwrapping the pointer. (Or just revert these pointer methods to the previous version)
But in this particular case, I think there is a design issue. These methods should be connected to something else, other than Privileges "struct". Do you have a container struct that holds these privileges?

Reply:
I bumped into this nil check issue during C++. What you suggest, would mean to have tons of nil checks one for each and every function. I know that is the safe way. But also learned, that it has to be considered whether it is really needed or we have a certain degree of confidence that the code always initializes the pointer. IN this case I think it would be over engineering putting a nil check.

Regarding to the question. Privileges is a list of Privilege structures. That only should be initialized once when the server loads the DB. So potentially the init can be inside the NewDBController() function. The intent of the IsValid functions is to have just a function call when checking the validity instead of doing a loop every time the code wants to validate. It has no state, so having a value receiver would make sense.

I will change it back to value receiver for now and later on we can think of a better solution. For an MVP this is just perfect as it is. :)